src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -442,6 +459,84 @@ async def validate(self, data, schema_name, verrors, old=None):
                     f"The following security flavor(s) require NFSv4 to be enabled: {','.join(v4_sec)}."
                 )
 
+    @private
+    def validate_and_clean_share_hosts_intput(self, hosts, schema_name, verrors):
+        """
+        The host field can contain multiple 'host' entries that are space delimited
+        The validation is limited to detecting repeated 'host' names
+
+        This will return a list of hosts
+        """
+        new_host_list = []
+        for host_str in hosts:
+            # Trap space delimited hosts
+            if ' ' in host_str:
+                new_host_list.extend(shlex.split(host_str))
```

Comment:
We should raise a ValidationError if users are trying to cram multiple addresses /  networks as a space-delimited list.